keras/metrics.py

```diff
@@ -34,7 +34,7 @@ def categorical_accuracy(y_true, y_pred):
 
 
 def sparse_categorical_accuracy(y_true, y_pred):
-    return K.cast(K.equal(K.max(y_true, axis=-1),
+    return K.cast(K.equal(y_true,
```

Comment:
Max might have been used when values are equal?

Reply:
equal is used to compare y_true with y_pred, it's still there. apply max to `y_true=[0, 1, 1]` will only result to `1`. which make no sense.  `y_true=[0, 1, 1]` means there are 3 data each with class 0, 1, and 1 as true labels